monai/bundle/config_item.py

```diff
@@ -286,9 +306,27 @@ def instantiate(self, **kwargs: Any) -> object:
             return None
 
         modname = self.resolve_module_name()
-        mode = self.get_config().get("_mode_", CompInitMode.DEFAULT)
+        mode = self.get_config().get(CONFIG_COMPONENT_KEY_MODE, CompInitMode.DEFAULT)
         args = self.resolve_args()
         args.update(kwargs)
+        wrapper = self.get_wrapper()
```

Comment:
I would split this code away into a separate class function as it takes a bit of the readability of the `instantiate` otherwise. 

Probably a `set_wrapper`, `get_wrapper` duo. 

```python

self.set_wrapper() # Sets wrapper if the _target_ instantiation is encountered in a _wrapper_ block. 
wrapper = self.get_wrapper()
.....


if wrapper is None:
    instantiate(modname, mode, **args)
else:
    wrapper(instantiate(modname, mode, **args))
```